v2/anomalydetection/frontend-setup.mdx

```diff
@@ -0,0 +1,129 @@
+
+Using it on other flows will not make a real difference, as, by their nature, other flows prevent this kind of anomalies.
+
+For detecting brute force attacks, password breaches, etc, you can skip these steps as the backend setup will handle this.
+
+### Generating a Request ID on the Frontend
+
+In order to generate a request ID, you need to import and initialise the SDK using your public API key. This SDK is used to generate a unique request ID for each authentication event attempt.
+
+Below is an example of how to implement request ID generation on your frontend:
+
+
+```tsx
+const API_KEY = "<api-key>"; // Your public API key that you received from the SuperTokens team
+const SDK_URL = "https://supertokens.com/JfNbWqtXUiGYuTqb/PYmK3PIPqdvHw6iz";
+
+const supertokensDeviceIdPromise = require(SDK_URL + "?apiKey=" + API_KEY).then((DeviceId: any) => DeviceId.load({
+    endpoint: [
+        SDK_URL,
+        DeviceId.defaultEndpoint
+    ]
+}));
+
+async function getRequestId() {
+    const deviceId = await supertokensDeviceIdPromise;
+    const result = await deviceId.get();
+    return result.requestId;
+}
+```
+
+### Passing the Request ID to the Backend
+
+Once you have generated the request ID on the frontend, you need to pass it to the backend. This is done by including the `requestId` property along with the value as part of the preAPIHook body from the initialisation of the recipes.
+
+:::important
+If the request ID is not passed to the backend, the anomaly detection will only be able to detect password breaches and brute force attacks.
+:::
+
+```tsx
+const preAPIHook = async (context) => {
+    let url = context.url;
+    let requestInit = context.requestInit;
+
+    let action = context.action;
+    
+    let requestId = await getRequestId();
+    let body = context.requestInit.body;
+    if (body !== undefined) {
+        let bodyJson = JSON.parse(body as string);
+        
+        bodyJson.requestId = requestId;
+        
+        requestInit.body = JSON.stringify(bodyJson);
+    }
+    
+    return {
+        requestInit, url
+    };
+}
+```
+
+Below is a full example of how to configure the SDK and pass the request ID to the backend:
+```tsx
+import EmailPassword from "supertokens-auth-react/recipe/emailpassword";
+
+const API_KEY = "<api-key>"; // Your public API key that you received from the SuperTokens team
+const SDK_URL = "https://supertokens.com/JfNbWqtXUiGYuTqb/PYmK3PIPqdvHw6iz";
+
+const supertokensDeviceIdPromise = require(SDK_URL + "?apiKey=" + API_KEY).then((DeviceId: any) => DeviceId.load({
+    endpoint: [
+        SDK_URL,
+        DeviceId.defaultEndpoint
+    ]
+}));
+
+async function getRequestId() {
+    const deviceId = await supertokensDeviceIdPromise;
+    const result = await deviceId.get();
+    return result.requestId;
+}
+
+const preAPIHook = async (context) => {
```

Comment:
not all requests should have this. Email password recipe has other types of api calls too. Only sign in, sign up and generating password reset token should have this

Reply:
done i think